pkg/scheduler/objects/queue.go

```diff
@@ -1242,9 +1240,6 @@ func (sq *Queue) internalGetMax(parentLimit *resources.Resource) *resources.Reso
 	defer sq.RUnlock()
 	// no parent queue limit set, not even for root
 	if parentLimit == nil {
-		if sq.maxResource == nil {
-			return nil
-		}
```

Comment:
I think we should keep the nil check to prevent panic error, even though it's not possible in the current production code. (root queue always have maxResource) 

Additionally, it seems that the nil checks for`parentLimit` and `sq.maxResource` in `internalGetMax()` are redundant, the resources.ComponentWiseMin(parentLimit, sq.maxResource) will do the same check.

https://github.com/wilfred-s/yunikorn-core/blob/a8537787fd20c65c56357d15a7219879d586c21c/pkg/scheduler/objects/queue.go#L1241-L1248

Reply:
No need for the check, as the Clone() method is nil-safe.